### PR TITLE
Directory search join button

### PR DESCRIPTION
**Pull request by dbkr**
Vector side of https://github.com/matrix-org/matrix-react-sdk/pull/496/files

Requires https://github.com/matrix-org/matrix-react-sdk/pull/496

### CODE REVIEW

**Comment by dbkr** [comment]
Tests now failing because it's not running with the react-sdk PR.

File: src/components/structures/RoomDirectory.js [hunk 1]
```diff
@@ -230,10 +231,22 @@ module.exports = React.createClass({
...
+        // update immediately
+        this.refreshRoomList();
+    },
+
+    onJoinClick: function(alias) {
+        this.showRoomAlias(alias);
+    },
+
+    collectSearchBox: function(e) {
+        this.searchBox = e;
```
Comment (richvdh) [inline src/components/structures/RoomDirectory.js]: what's this for?

**Comment by dbkr** [comment]
ptal

File: src/components/structures/RoomDirectory.js [hunk 2]
```diff
@@ -393,13 +401,15 @@ module.exports = React.createClass({
...
+        const DirectorySearchBox = sdk.getComponent('elements.DirectorySearchBox');
         return (
             <div className="mx_RoomDirectory">
                 <SimpleRoomHeader title="Directory" />
                 <div className="mx_RoomDirectory_list">
                     <div className="mx_RoomDirectory_listheader">
-                        <input type="text" placeholder="Find a room by keyword or room ID (#foo:matrix.org)"
-                            className="mx_RoomDirectory_input" size="64" onChange={this.onFilterChange} onKeyUp={this.onFilterKeyUp}
+                        <DirectorySearchBox
+                            className="mx_RoomDirectory_searchbox" ref={this.collectSearchBox}
```
Comment (richvdh) [inline src/components/structures/RoomDirectory.js]: redundant `ref`

**Pull request merged**